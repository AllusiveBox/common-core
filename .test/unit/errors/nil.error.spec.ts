import "jest-extended";
import { NilError } from "../../../.src";

/**
 *
 * Test Suite for the NilError class.
 *
 * @group unit
 * @group errors
 * @group nil
 *
 */
describe(
    "NilError Unit Test Suite",
    () => {

        test(
            "that NilError initializes correctly",
            () => {
                // setup
                const error = new NilError();

                expect(error)
                    .toMatchObject({
                        cause: "Null or undefined value",
                        context: undefined,
                        message: "Null or undefined value",
                        name: "NilError"
                    });
            }
        );

        test.each([
            null,
            undefined
        ])
        (
            "that given %s, NilError correctly sets the message field",
            (arg) => {
                // Setup
                const error: NilError<never> = new NilError<never>(arg as any);

                expect(error.message)
                    .toMatchSnapshot();
            }
        );

        test(
            "that given a message, NilError correctly sets the message field",
            () => {
                // Setup
                const error: NilError<never> = new NilError<never>("Test");

                expect(error.message)
                    .toMatchInlineSnapshot(`"Test"`);
            }
        );

    }
);
