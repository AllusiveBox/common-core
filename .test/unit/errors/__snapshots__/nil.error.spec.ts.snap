// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NilError Unit Test Suite that given null, NilError correctly sets the message field 1`] = `"Null or undefined value"`;

exports[`NilError Unit Test Suite that given undefined, NilError correctly sets the message field 1`] = `"Null or undefined value"`;
